starter-code/src/functions-and-arrays.js

```diff
@@ -98,3 +140,19 @@ var matrix = [
   [20, 73, 35, 29, 78, 31, 90, 1, 74, 31, 49, 71, 48, 86, 81, 16, 23, 57, 5, 54],
   [1, 70, 54, 71, 83, 51, 54, 69, 16, 92, 33, 48, 61, 43, 52, 1, 89, 19, 67, 48]
 ];
+
+function greatestProduct(myArray){
```

Comment:
Your bonus doesn't quiet work- this part is hard and the instructions are not super clear. You need to loop of the matrix (a for loop inside a for loop is the easiest way to do this) 

by returning 16 directly is cheating the tests meaning it won't work if Jasmine changes the input.

I'm happy to review this part with you, so you can speak to me (Abi) during a break or lunch, or I can send you a copy of the solution code. But that depends on if you want to try again on your own first